pkg/settings/recording/config.go

```diff
@@ -20,12 +23,20 @@ func (cfg *Config) RegisterFlags(fs *flag.FlagSet) {
 		false,
 		"Enable the storing of recording rules in tenant settings.",
 	)
+	fs.StringVar(
+		&cfg.Address,
+		flagAddress,
+		"localhost:4040",
+		"The address of the client for the tenant-settings",
+	)
 }
 
 func (cfg *Config) Validate() error {
 	if !cfg.Enabled {
 		return nil
 	}
-
+	if cfg.Address == "" {
+		return fmt.Errorf("%s is required", flagAddress)
```

Comment:
I don't think this flag should be required to be set when the tenant settings are running, but they maybe should be required when the compaction workers is enabled with metric exporter.